src/components/ProgressBar.tsx

```diff
@@ -0,0 +1,79 @@
+import React, {useEffect} from 'react';
+import Animated, {cancelAnimation, Easing, runOnJS, useAnimatedStyle, useSharedValue, withDelay, withRepeat, withSequence, withTiming} from 'react-native-reanimated';
+import useThemeStyles from '@hooks/useThemeStyles';
+import CONST from '@src/CONST';
+
+function ProgressBar({shouldShow}: {shouldShow: boolean}) {
+    const left = useSharedValue(0);
+    const width = useSharedValue(0);
+    const opacity = useSharedValue(0);
+    const isVisible = useSharedValue(false);
+    const styles = useThemeStyles();
+
+    useEffect(() => {
+        if (shouldShow) {
+            // eslint-disable-next-line react-compiler/react-compiler
+            isVisible.value = true;
+            left.value = 0;
+            width.value = 0;
+            opacity.value = withTiming(1, {duration: CONST.ANIMATED_PROGRESS_BAR_OPACITY_DURATION});
+            left.value = withDelay(
+                CONST.ANIMATED_PROGRESS_BAR_DELAY,
+                withRepeat(
+                    withSequence(
+                        withTiming(0, {duration: 0}),
+                        withTiming(0, {duration: CONST.ANIMATED_PROGRESS_BAR_DURATION, easing: Easing.bezier(0.65, 0, 0.35, 1)}),
+                        withTiming(100, {duration: CONST.ANIMATED_PROGRESS_BAR_DURATION, easing: Easing.bezier(0.65, 0, 0.35, 1)}),
+                    ),
+                    -1,
+                    false,
+                ),
+            );
+
+            width.value = withDelay(
+                CONST.ANIMATED_PROGRESS_BAR_DELAY,
+                withRepeat(
+                    withSequence(
+                        withTiming(0, {duration: 0}),
+                        withTiming(100, {duration: CONST.ANIMATED_PROGRESS_BAR_DURATION, easing: Easing.bezier(0.65, 0, 0.35, 1)}),
+                        withTiming(0, {duration: CONST.ANIMATED_PROGRESS_BAR_DURATION, easing: Easing.bezier(0.65, 0, 0.35, 1)}),
+                    ),
+                    -1,
+                    false,
+                ),
+            );
+        } else if (isVisible.value) {
+            opacity.value = withTiming(0, {duration: CONST.ANIMATED_PROGRESS_BAR_OPACITY_DURATION}, () => {
+                runOnJS(() => {
+                    isVisible.value = false;
+                    cancelAnimation(left);
+                    cancelAnimation(width);
+                });
+            });
+        }
+        // eslint-disable-next-line react-compiler/react-compiler, react-hooks/exhaustive-deps
+    }, [shouldShow]);
+
+    const animatedIndicatorStyle = useAnimatedStyle(() => {
+        return {
+            left: `${left.value}%`,
+            width: `${width.value}%`,
+        };
+    });
+
+    const animatedContainerStyle = useAnimatedStyle(() => {
+        return {
+            opacity: opacity.value,
```

Comment:
Any way to hide the container? i can see it with a white background when it isn't active

![Screenshot 2024-10-01 at 12 24 36 in the afternoon](https://github.com/user-attachments/assets/d73e118c-dece-463e-b966-d52cf5f043d6)

Reply:
From this [comment](https://github.com/Expensify/App/pull/48772#issuecomment-2385348856), Do I need to fix this?

Reply:
I think so yes, looks off to leave it like that. WDYT @dubielzyk-expensify?

Reply:
@getusha i fixed